source/app/controllers/sessions_controller.rb

```diff
@@ -1,12 +1,39 @@
 class SessionsController < ApplicationController
 
+  def new
+    @user = User.new
+  end
+
   def index
   end
 
   def create
+    @user = User.find_by_email(params[:email])
+    if @user.nil?
+      flash[:login_error] = 'Login Error occured'
+      redirect_to '/' and return
```

Comment:
I prefer two lines for this (even though it isn't the common Rails convention).

```ruby
  redirect_to '/'
  return
```

`and return` implies the `return` is conditional based on `redirect_to` returning something truthy. It does, but that's not the point. If Rails changed `redirect_to` to return `nil`, we still want to `return` from the method here.

`redirect_to '/'; return` conveys more accurately that we want these two commands invokes, but we tend to put separate commands on their own lines.